File: SpringCodingContest/I.py
q = int(input())
a = int(input())

answers = [1]*(10**6)

for i in range(2,len(answers)):
    for j in range(2*i,len(answers),i):
        answers[j] += i

solution = 0
for i in range(q):
    solution += answers[a]
    a = ((a*a) % 999983) + 1

print(solution)

File: SpringCodingContest/I.cpp
#include <iostream>

typedef long long LL;

#define MAXQ 1000010

LL answers[MAXQ];
int main() {
    LL q;
    std::cin >> q;

    LL a;
    std::cin >> a;

    for (int i = 1; i < MAXQ; i++) {
        for (int j = 2*i; j < MAXQ; j += i) {
            answers[j] += i;
        }
    }

    LL solution = 0;
    for (int i = 0; i < q; i++) {
        solution += answers[a];
        a = ((a*a) % 999983) + 1;
    }

    std::cout << solution << std::endl;
}
